TechnicalDocuments/Proposals/StyleSheets.md

```diff
@@ -0,0 +1,119 @@
+# Composable value-type style sheets
+
+* Author(s): Anders Ha
+* Review Manager: Sergey Shulga, David Rodrigues
+
+## Problem Statement
+The current inheritance based StyleSheet approach is unideal in three areas:
+
+1. Ease of Use: Requiring upfront declarations of all styleable properties.
+
+    We would have to constantly match UIKit changes. Custom views would require manual implementation.
+
+2. Mental Model: Awkwardness when being applied to views that are opened for extension.
+
+    When one defines a non generic `ExtensibleViewStyleSheet` for an `ExtensibleView`, the `StyleSheetProtocol.Element` associated type is immediately satisfied with the concrete view type. This means in a generic environment bound to `StyleSheetProtocol`, only `apply(to _: ExtensibleView)` would be recognised since `Element == ExtensibleView` per the conformance. In other words, all stylesheet subclasses must override this base class typed method & are required to downcast at runtime to access stuff defined in the subclass.
+
+    On the other hand, when one defines a generic `ExtensibleViewStyleSheet<T: ExtensibleView>` stylesheet, it does address the said issue by satisfying `StyleSheetProtocol.Element` with a generic parameter. However, it leads to awkward and verbose spellings at use sites in general, e.g. `ExtensibleViewStyleSheet<ExtensibleView>`.
+
+3. Correctness: Unintended Sharing
+
+    Stylesheets are currently classes so as to be able to take advantage of inheritance, and this makes them prone to unintended sharing. With Swift offering first-class value type, this might amplify its possibility due to common perception of configurations being modelled as value types.
```

Comment:
```suggestion
    Stylesheets are currently classes so as to be able to take advantage of inheritance. Since there is a common perception in Swift of configurations being modelled as value types this creates a risk of unintended sharing
```